src/main/webapp/app/iris/base-chatbot/iris-base-chatbot.component.ts

```diff
@@ -0,0 +1,371 @@
+            }
+        }
+    }
+
+    /**
+     * Handles the input event in the message textarea.
+     */
+    onInput() {
+        this.adjustTextareaRows();
+    }
+
+    /**
+     * Handles the paste event in the message textarea.
+     */
+    onPaste() {
+        setTimeout(() => {
+            this.adjustTextareaRows();
+        }, 0);
+    }
+
+    /**
+     * Adjusts the height of the message textarea based on its content.
+     */
+    adjustTextareaRows() {
+        const textarea: HTMLTextAreaElement = this.messageTextarea.nativeElement;
+        textarea.style.height = 'auto'; // Reset the height to auto
+        const lineHeight = parseInt(getComputedStyle(textarea).lineHeight, 10) + 4;
+        const maxRows = 3;
+        const maxHeight = lineHeight * maxRows;
+
+        textarea.style.height = `${Math.min(textarea.scrollHeight, maxHeight)}px`;
+
+        this.adjustScrollButtonPosition(Math.min(textarea.scrollHeight, maxHeight) / lineHeight);
+    }
+
+    /**
+     * Handles the row change event in the message textarea.
+     */
+    onModelChange() {
+        const textarea: HTMLTextAreaElement = this.messageTextarea.nativeElement;
+        const newRows = textarea.value.split('\n').length;
+        if (newRows != this.rows) {
+            if (newRows <= 3) {
+                textarea.rows = newRows;
+                this.adjustScrollButtonPosition(newRows);
+                this.rows = newRows;
+            }
+        }
+    }
+
+    /**
+     * Adjusts the position of the scroll button based on the number of rows in the message textarea.
+     * @param newRows - The new number of rows.
+     */
+    adjustScrollButtonPosition(newRows: number) {
+        const textarea: HTMLTextAreaElement = this.messageTextarea.nativeElement;
+        const scrollArrow: HTMLElement = this.scrollArrow.nativeElement;
+        const lineHeight = parseInt(window.getComputedStyle(textarea).lineHeight);
+        const rowHeight = lineHeight * newRows - lineHeight;
+        setTimeout(() => {
+            scrollArrow.style.bottom = `calc(11% + ${rowHeight}px)`;
+        }, 10);
+    }
+
+    /**
+     * Resets the height of the chat body.
+     */
+    resetChatBodyHeight() {
+        const textarea: HTMLTextAreaElement = this.messageTextarea.nativeElement;
+        const scrollArrow: HTMLElement = this.scrollArrow.nativeElement;
+        textarea.rows = 1;
+        textarea.style.height = '';
+        scrollArrow.style.bottom = '';
+    }
+
+    checkChatScroll() {
+        const messagesElement = this.messagesElement.nativeElement;
+        const scrollTop = messagesElement.scrollTop;
+        this.isScrolledToBottom = scrollTop < 50;
+    }
+
+    protected readonly IrisLogoSize = IrisLogoSize;
+    protected readonly IrisMessageContentType = IrisMessageContentType;
+    protected readonly IrisAssistantMessage = IrisAssistantMessage;
+    protected readonly IrisTextMessageContent = IrisTextMessageContent;
+    protected readonly JSON = JSON;
```

Comment:
Not necessary to fix now, please move these constants to the top.

Reply:
Something we should discuss since its only boilerplate and reduces readability, and my IntelliJ adds it automatically to the end of the file, but ok for now